usr/opt/OSC200/usr/share/man/man5/ascii.5
.\" This manpage source uses rsml coding. 
.so /usr/share/lib/tmac/sml
.so /usr/share/lib/tmac/rsml
.\"
.\" 
.\" *********************************************************************
.\" *                                                                   *
.\" *      Copyright (c) Digital Equipment Corporation, 1991, 1994      *
.\" *                                                                   *
.\" *                       All Rights Reserved.                        *
.\" *                                                                   *
.\" *********************************************************************
.\"
.\"
.\" HISTORY
.\"
.\" @(#)$RCSfile: ascii.5,v $ $Revision: 3.2.5.3 $ (DEC) $Date: 1993/07/16 19:36:02 $
.\"
.\" 
.\" (c) Copyright 1991, OPEN SOFTWARE FOUNDATION, INC.
.\" ALL RIGHTS RESERVED
.\" 
.\" 
.\" Copyright (c) 1989, 1990 The Regents of the University of California.
.\" All rights reserved.
.\" 
.\" Redistribution and use in source and binary forms are permitted provided
.\" that: (1) source distributions retain this entire copyright notice and
.\" comment, and (2) distributions including binaries display the following
.\" acknowledgement:  ``This product includes software developed by the
.\" University of California, Berkeley and its contributors'' in the
.\" documentation or other materials provided with the distribution and in
.\" all advertising materials mentioning features or use of this software.
.\" Neither the name of the University nor the names of its contributors may
.\" be used to endorse or promote products derived from this software without
.\" specific prior written permission.
.\" THIS SOFTWARE IS PROVIDED ``AS IS'' AND WITHOUT ANY EXPRESS OR IMPLIED
.\" WARRANTIES, INCLUDING, WITHOUT LIMITATION, THE IMPLIED WARRANTIES OF
.\" MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
.\" 
.\" @(#)ascii.7	6.3 (Berkeley) 6/24/90
.\" 
.TH ascii 5
.SH NAME
.PP
\*Lascii\*O \- Octal, hexadecimal, and decimal ASCII character sets
.SH DESCRIPTION
.PP
.iX "ascii character set"
.iX "character set" "ascii"
.\" BEGIN Digital-specific information
.\"
.\" The three tables have been reformatted to make them 8 columns wide
.\" for an intelligible display.  No technical changes have been made.
.\" - rfb 9307.16
The
octal
character
set is:
.oS
\*L000 nul  020 dle  040 sp   060 0    100 @    120 P    140 \`    160 p
001 soh  021 dc1  041 !    061 1    101 A    121 Q    141 a    161 q
002 stx  022 dc2  042 "    062 2    102 B    122 R    142 b    162 r
003 etx  023 dc3  043 #    063 3    103 C    123 S    143 c    163 s
004 eot  024 dc4  044 $    064 4    104 D    124 T    144 d    164 t
005 enq  025 nak  045 %    065 5    105 E    125 U    145 e    165 u
006 ack  026 syn  046 &    066 6    106 F    126 V    146 f    166 v
007 bel  027 etb  047 '    067 7    107 G    127 W    147 g    167 w
010 bs   030 can  050 (    070 8    110 H    130 X    150 h    170 x
011 ht   031 em   051 )    071 9    111 I    131 Y    151 i    171 y
012 nl   032 sub  052 *    072 :    112 J    132 Z    152 j    172 z
013 vt   033 esc  053 +    073 ;    113 K    133 [    153 k    173 {
014 np   034 fs   054 ,    074 <    114 L    134 \e    154 l    174 |
015 cr   035 gs   055 -    075 =    115 M    135 ]    155 m    175 }
016 so   036 rs   056 .    076 >    116 N    136 ^    156 n    176 ~
017 si   037 us   057 /    077 ?    117 O    137 _    157 o    177 del\*O
.oE
.PP
The
hexadecimal
character
set is:
.oS
\*L 00 nul   10 dle   20 sp    30 0     40 @     50 P     60 \`     70 p
 01 soh   11 dc1   21 !     31 1     41 A     51 Q     61 a     71 q
 02 stx   12 dc2   22 "     32 2     42 B     52 R     62 b     72 r
 03 etx   13 dc3   23 #     33 3     43 C     53 S     63 c     73 s
 04 eot   14 dc4   24 $     34 4     44 D     54 T     64 d     74 t
 05 enq   15 nak   25 %     35 5     45 E     55 U     65 e     75 u
 06 ack   16 syn   26 &     36 6     46 F     56 V     66 f     76 v
 07 bel   17 etb   27 '     37 7     47 G     57 W     67 g     77 w
 08 bs    18 can   28 (     3f ?     48 H     58 X     68 h     78 x
 09 ht    19 em    29 )     38 8     49 I     59 Y     69 i     79 y
 0a nl    1a sub   2a *     39 9     4a J     5a Z     6a j     7a z
 0b vt    1b esc   2b +     3a :     4b K     5b [     6b k     7b {
 0c np    1c fs    2c ,     3b ;     4c L     5c \e     6c l     7c |
 0d cr    1d gs    2d -     3c <     4d M     5d ]     6d m     7d }
 0e so    1e rs    2e .     3d =     4e N     5e ^     6e n     7e ~
 0f si    1f us    2f /     3e >     4f O     5f _     6f o     7f del\*O
.oE
.PP
The
decimal
character
set is:
.PP
.oS
\*L  0 nul   16 dle   32 sp    48 0     64 @     80 P     96 \`    112 p
  1 soh   17 dc1   33 !     49 1     65 A     81 Q     97 a    113 q
  2 stx   18 dc2   34 "     50 2     66 B     82 R     98 b    114 r
  3 etx   19 dc3   35 #     51 3     67 C     83 S     99 c    115 s
  4 eot   20 dc4   36 $     52 4     68 D     84 T    100 d    116 t
  5 enq   21 nak   37 %     53 5     69 E     85 U    101 e    117 u
  6 ack   22 syn   38 &     54 6     70 F     86 V    102 f    118 v
  7 bel   23 etb   39 '     55 7     71 G     87 W    103 g    119 w
  8 bs    24 can   40 (     56 8     72 H     88 X    104 h    120 x
  9 ht    25 em    41 )     57 9     73 I     89 Y    105 i    121 y
 10 nl    26 sub   42 *     58 :     74 J     90 Z    106 j    122 z
 11 vt    27 esc   43 +     59 ;     75 K     91 [    107 k    123 {
 12 np    28 fs    44 ,     60 <     76 L     92 \e    108 l    124 |
 13 cr    29 gs    45 -     61 =     77 M     93 ]    109 m    125 }
 14 so    30 rs    46 .     62 >     78 N     94 ^    110 n    126 ~
 15 si    31 us    47 /     63 ?     79 O     95 _    111 o    127 del\*O
.oE
.\" END Digital-specific information
.EQ
delim off
.EN


usr/opt/OSC200/src/usr/ccs/lib/libiconv/ascii.c
/*
 * *****************************************************************
 * *                                                               *
 * *    Copyright (c) Digital Equipment Corporation, 1991, 1994    *
 * *                                                               *
 * *   All Rights Reserved.  Unpublished rights  reserved  under   *
 * *   the copyright laws of the United States.                    *
 * *                                                               *
 * *   The software contained on this media  is  proprietary  to   *
 * *   and  embodies  the  confidential  technology  of  Digital   *
 * *   Equipment Corporation.  Possession, use,  duplication  or   *
 * *   dissemination of the software and media is authorized only  *
 * *   pursuant to a valid written license from Digital Equipment  *
 * *   Corporation.                                                *
 * *                                                               *
 * *   RESTRICTED RIGHTS LEGEND   Use, duplication, or disclosure  *
 * *   by the U.S. Government is subject to restrictions  as  set  *
 * *   forth in Subparagraph (c)(1)(ii)  of  DFARS  252.227-7013,  *
 * *   or  in  FAR 52.227-19, as applicable.                       *
 * *                                                               *
 * *****************************************************************
 */
/*
 * HISTORY
 */
#ifndef lint
static char	*sccsid = "@(#)$RCSfile: ascii.c,v $ $Revision: 1.1.6.2 $ (DEC) $Date: 1993/06/08 00:38:46 $";
#endif
/*
 * (c) Copyright 1990, 1991, 1992, 1993 OPEN SOFTWARE FOUNDATION, INC. 
 * ALL RIGHTS RESERVED 
 */
/*
 * OSF/1 1.2
 */
/*
 * COMPONENT_NAME: (LIBICONV)
 *
 * FUNCTIONS:
 *
 *
 * (C) COPYRIGHT International Business Machines Corp. 1991
 * All Rights Reserved
 * Licensed Materials - Property of IBM
 *
 * US Government Users Restricted Rights - Use, duplication or
 * disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 * 
 * 1.3  com/lib/iconv/ascii.c, libiconv, bos320, 9130320 7/16/91 03:05:12
 */

#include <stdio.h>
#include <string.h>
#include <sys/types.h>
#include "iconv_local.h"

static char	valid[256] = {
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,

	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
	1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,

	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,

	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
	0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
};

int	_ascii_exec(unsigned char** inbuf, size_t *inbytesleft, 
	unsigned char** outbuf, size_t *outbytesleft)
{
	unsigned char 	*in, *e_in;		/* point inbut buffer */
	unsigned char 	*out, *e_out;		/* point output buffer */
	int	ret_value;

	e_in = (in = *inbuf) + *inbytesleft;
	e_out = (out = *outbuf) + *outbytesleft;
	while (1) {
		if (in >= e_in) {
			ret_value = ICONV_DONE;
			break;
		}
		if (out >= e_out) {
			ret_value = ICONV_OVER;
			break;
		}
		if (!valid[*in]) {
			ret_value = ICONV_INVAL;
			break;
		}
		*out++ = *in++;
	}
	*inbuf = in;
	*inbytesleft = e_in - in;
	*outbuf = out;
	*outbytesleft = e_out - out;
	return ret_value;
}
